### DISCUSSION TITLE
Main stream of sub stream for detection

### DISCUSSION THREAD

**Discussion by chokays** [blakeblackshear/frigate#2345] [Ask A Question] [answered]
My Hikvision DS-2CD2385FWD has two streams:
Main stream max 3840 x 2160
Sub stream max 640 x 480

Looking at the installation guide, it seems the max resolution of my sub stream might be too low for proper detection?

Should I use the main stream instead? If so, I would prefer to use the main stream at the max resolution of 3840 x 2160.

**Comment by Minglarn** [comment]
Use the main stream... I am.. And it works better (at least for me) and the snapshots are more detailed..

**Comment by blakeblackshear** [comment, marked as answer]
> Should I use the main stream instead? If so, I would prefer to use the main stream at the max resolution of 3840 x 2160

You can also set a lower resolution in your detect config such as 1080p. This will move some of the CPU load from frigate's code to ffmpeg and may be preferable. It will reduce the snapshot resolution. I would also recommend setting your frame_height to 50 and your contour_area to 30 in your motion config to reduce CPU load.
